packages/react-query/src/useBaseQuery.ts

```diff
@@ -131,7 +131,7 @@ export function useBaseQuery<
   }
 
   // Handle result property usage tracking
-  return !defaultedOptions.notifyOnChangeProps
+  return React.useMemo(() => !defaultedOptions.notifyOnChangeProps
     ? observer.trackResult(result)
-    : result
+    : result, [defaultedOptions.notifyOnChangeProps, observer.trackResult, result])
```

Comment:
`result` comes from `observer.getOptimisticResult`:

https://github.com/TanStack/query/blob/34ca5581217dae4133046f6cfbc746a22be14dd9/packages/react-query/src/useBaseQuery.ts#L81

which is re-computed on every render and will yield a new object every time. So this memoization does nothing in practice I'm afraid.